src/radiant.c

```diff
@@ -684,6 +689,9 @@ int radiant_read_event(radiant_dev_t * bd, rno_g_header_t * hd, rno_g_waveform_t
   wf->run_number = bd->run; 
 
   wf->radiant_nsamples = nsamples;
+  
```

Comment:
does this really need to happen on every event? Or can it be cached? On the DAQ, this will cause contention between the monitor and acq threads, as otherwise, I believe radiant_read_event only uses the SPI bus (no UART). In fact, looking at rno-g-acq, with the current locking scheme, this will almost certainly cause crashes as both threads attempt to access the UART. 

I would suggest that this value be cached on radiant_open and updated on setting. It shouldn't change otherwise anyway...

Reply:
Only needs to happen once before taking data or at the start of a run. I knew that the uart read for each event wouldn't be the best but wasn't sure how to handle it.

Reply:
Yeah, I would just store it inside struct radiant_dev, have it filled in on radiant_open and then update it on every radiant_set_delays. Then radiant_get_delays would just copy from struct radiant_dev. Since radiant_dev_t is opaque, you can add anything to it without worrying about API/ABI.